### PR TITLE
Add systemd service file

### PR DESCRIPTION
**Pull request by mariogrip**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,12 +34,27 @@ add_subdirectory(src)
...
 
+# Systemd file
+pkg_check_modules(SYSTEMD systemd)
+if (${SYSTEMD_FOUND})
+  pkg_get_variable(SYSTEMD_SYSTEM_DIR systemd systemdsystemunitdir)
+  message (STATUS "${SYSTEMD_SYSTEM_DIR} is the systemd system unit file install dir, installing ${systemd_file} to it")
+  install (FILES "${systemd_file}"
+           DESTINATION "${SYSTEMD_SYSTEM_DIR}")
+else()
+   message(WARNING "Did not find systmed, not installing systemd service")
```
Comment (PureTryOut) [inline CMakeLists.txt:54]: Typo "systmed"

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,12 +34,27 @@ add_subdirectory(src)
...
 add_subdirectory(qt)
 
-# Dbus file
-install(FILES data/hfd-service.conf 
+FILE(GLOB upstart_file "${CMAKE_CURRENT_SOURCE_DIR}/data/hfd-service.conf")
+FILE(GLOB systemd_file "${CMAKE_CURRENT_SOURCE_DIR}/data/hfd-service.service")
+FILE(GLOB dbus_file "${CMAKE_CURRENT_SOURCE_DIR}/data/com.lomiri.hfd.conf")
+
+# Upstart file
+install(FILES ${upstart_file}
```
Comment (PureTryOut) [inline CMakeLists.txt:42]: Maybe add an option to not install the upstart file at all? I don't think there are many people that have a need for it

**Review by PureTryOut — COMMENTED** [review]
I'm thinking, I personally want an OpenRC service file but there is no way to check from CMake if OpenRC is installed (afaik), how would we make installing that file optional in the same way? Normally I'd just add a `option(OPENRC "Install OpenRC service file" OFF)` or something that packaging can turn on.

**Comment by peat-psuwit** [comment]
Superseded by https://gitlab.com/ubports/core/hfd-service/-/merge_requests/2.

**Pull request closed without merging**